addons/html_editor/static/src/main/hint_plugin.js

```diff
@@ -66,34 +66,33 @@ export class HintPlugin extends Plugin {
      */
     updateHints(root = this.editable) {
         this.clearHints(root);
-        this.makeEmptyBlockHints(root);
         this.updateTempHint(this.shared.getEditableSelection());
     }
 
-    makeEmptyBlockHints(root) {
-        for (const { selector, hint } of this.resources.emptyBlockHints) {
-            for (const el of selectElements(root, selector)) {
-                // @todo: consider using isEmptyBlock instead.
-                if (isEmpty(el) && !isProtected(el) && !isProtecting(el)) {
-                    this.makeHint(el, hint);
-                }
-            }
-        }
-    }
-
     updateTempHint(selection) {
         if (this.tempHint) {
             this.removeHint(this.tempHint);
         }
 
         if (selection.isCollapsed) {
             for (const hint of this.resources["temp_hints"]) {
-                const target = hint.target(selection, this.editable);
-                // Do not replace an existing empty block hint by a temp hint.
-                if (target && !target.classList.contains("o-we-hint")) {
-                    this.makeHint(target, hint.text);
-                    this.tempHint = target;
-                    return;
+                if (hint.selector) {
+                    const el = closestElement(selection.anchorNode, hint.selector);
+                    if (el && !isProtected(el) && isEmptyBlock(el)) {
```

Comment:
Here you changed `isEmpty` to `isEmptyBlock` (see my other comment below), but you also removed `isProtecting` from the condition. Was this intended? It looks like an oversight.

Reply:
@dmo-odoo `isProtecting` is already checked in `isEmptyBlock`, so there's no need to add it again here.